RULES_DESCRIPTIONS.md

```diff
@@ -921,6 +922,50 @@ More [information here](https://go.dev/wiki/CodeReviewComments#package-comments)
 
 _Configuration_: N/A
 
+## package-directory-mismatch
+
+_Description_: It is considered a good practice to name a package after the directory containing the file.
+This rule warns when the package name declared in the file does not match the name of the directory containing the file.
+
+The following cases are excluded from this check:
+
+- Package `main` (executable packages)
+- Test files (files with `_test` suffix)
+- Files in `testdata` directories (at any level) - by default
+- Files directly in `internal` directories (but files in subdirectories of `internal` are checked)
+- Files in version directories (`v1`, `v2`, etc.) - the parent directory name is used for comparison
```

Comment:
There are packages with literally v1/v2 names (also e.g. v1alpha1). I'd first attempt to match them only if no match, do what you propose here.